packages/camera/lib/camera.dart

```diff
@@ -0,0 +1,278 @@
+      : super(const CameraValue.uninitialized());
+
+  /// The size of the preview in pixels.
+  ///
+  /// Is [null] until [initialize] completes.
+  Size get previewSize => _previewSize;
+
+  /// Convenience getter for `previewSize.height / previewSize.width`.
+  ///
+  /// Can only be called after [initialize] completes
+  double get aspectRatio => previewSize.height / previewSize.width;
+
+  /// Initializes the camera on the device.
+  ///
+  /// Throws a [CameraException] if the initialization fails.
+  Future<Null> initialize() async {
+    if (_disposed) {
+      return;
+    }
+    try {
+      _creatingCompleter = new Completer<Null>();
+      final Map<String, dynamic> reply = await _channel.invokeMethod(
+        'create',
+        <String, dynamic>{
+          'cameraName': description.name,
+          'resolutionPreset': serializeResolutionPreset(resolutionPreset),
+        },
+      );
+      _textureId = reply['textureId'];
+      _previewSize = new Size(
+        reply['previewWidth'].toDouble(),
+        reply['previewHeight'].toDouble(),
+      );
+      value = value.copyWith(initialized: true);
+      _applyStartStop();
+    } on PlatformException catch (e) {
+      value = value.copyWith(errorDescription: e.message);
+      throw new CameraException(e.code, e.message);
+    }
+    _eventSubscription =
+        new EventChannel('flutter.io/cameraPlugin/cameraEvents$_textureId')
+            .receiveBroadcastStream()
+            .listen(_listener);
+    _creatingCompleter.complete(null);
+  }
+
+  void _listener(Map<String, dynamic> event) {
+    if (event['eventType'] == 'error') {
+      value = value.copyWith(errorDescription: event['errorDescription']);
+    }
+  }
+
+  /// Captures an image and saves it to [path].
+  ///
+  /// A path can for example be obtained using
+  /// [path_provider](https://pub.dartlang.org/packages/path_provider).
+  ///
+  /// Throws a [CameraException] if the capture fails.
+  Future<Null> capture(String path) async {
+    if (!value.initialized || _disposed) {
+      throw new CameraException(
+        'Uninitialized capture()',
+        'capture() was called on uninitialized CameraController',
+      );
+    }
+    try {
+      await _channel.invokeMethod(
+        'capture',
+        <String, dynamic>{'textureId': _textureId, 'path': path},
+      );
+    } on PlatformException catch (e) {
+      throw new CameraException(e.code, e.message);
+    }
+  }
+
+  void _applyStartStop() {
+    if (value.initialized && !_disposed) {
+      if (value.isStarted) {
+        _channel.invokeMethod(
+          'start',
+          <String, dynamic>{'textureId': _textureId},
+        );
+      } else {
+        _channel.invokeMethod(
+          'stop',
+          <String, dynamic>{'textureId': _textureId},
+        );
+      }
+    }
+  }
+
+  /// Starts the preview.
+  ///
+  /// If called before [initialize] it will take effect just after
+  /// initialization is done.
+  Future<Null> start() async {
```

Comment:
Why is this asynchronous? Did you intend to make `_applyStartStop` async too?

Reply:
Should not be async.